bpf/cpu/cpu.bpf.c

```diff
@@ -747,15 +815,28 @@ int walk_user_stacktrace_impl(struct bpf_perf_event_data *ctx) {
       break;
     }
 
+    //LOG("[debug] Switching to mixed-mode unwinding? %d", unwind_state->jitted);
+
     // Add address to stack.
     u64 len = unwind_state->stack.len;
     // Appease the verifier.
     // For some reason bailing out here if the condition is not true does
     // not work?
-    if (len >= 0 && len < MAX_STACK_DEPTH) {
-      unwind_state->stack.addresses[len] = unwind_state->ip;
+
+    // This is for the case when we are NOT switching unwinding from JIT to DWARF section
+    // i.e. unwind_state->jitted holds false
+    if (!unwind_state->jitted) {
+      if (len >= 0 && len < MAX_STACK_DEPTH) {
+        unwind_state->stack.addresses[len] = unwind_state->ip;
+
+        unwind_state->stack.len++;
+      }
     }
 
+    // Set unwind_state->jitted to false once we have checked for switch from JITed unwinding to DWARF unwinding
+    unwind_state->jitted = false;
+    //LOG("[debug] Switched to mixed-mode DWARF unwinding? %d", unwind_state->jitted);
```

Comment:
nit: stray comment?

Reply:
more like an explanation for future sumera who is very likely to get confused haha 😅 
(also ditto for above)

unless you are super against it, I would love to keep them :D